src/fides/api/models/privacy_notice.py

```diff
@@ -332,29 +351,152 @@ def create(
         data: dict[str, Any],
         check_name: bool = False,
     ) -> PrivacyNotice:
+        """
+        Creates a Privacy Notice and then a NoticeTranslation record for each supplied Translation.
+
+        For each Translation a historical record is created which combines details from the Notice and
+        the Translation for auditing purposes.  Privacy preferences are saved against the PrivacyNoticeHistory
+        record which contains the full details."""
+        translations = data.pop("translations", []) or []
         created = super().create(db=db, data=data, check_name=check_name)
 
-        # create the history after the initial object creation succeeds, to avoid
-        # writing history if the creation fails and so that we can get the generated ID
-        data.pop("id", None)
-        history_data = {**data, "privacy_notice_id": created.id}
-        PrivacyNoticeHistory.create(db, data=history_data, check_name=False)
+        for translation_data in translations:
+            data.pop(
+                "id", None
+            )  # Default Notices have id specified but we don't want to use the same id for the historical record
+            translation = NoticeTranslation.create(
+                db, data={**translation_data, "privacy_notice_id": created.id}
+            )
+            PrivacyNoticeHistory.create(
+                db,
+                data={**data, **translation_data, "translation_id": translation.id},
+                check_name=False,
+            )
+
         return created
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyNotice:
         """
-        Overrides the base update method to automatically bump the version of the
-        PrivacyNotice record and also create a new PrivacyNoticeHistory entry
+        Updates the Privacy Notice
+        - Upserts or deletes supplied translations to match translations in the request
+        - For each remaining translation, create a historical record if the base notice or translation changed.
```

Comment:
Privacy Notice update method has similar changes.  Multiple resources in other tables are being upserted here, need to revisit with the best way to do this.